Update to Ice & Fire 2.0

kotlin-programmer opened the pull request:
These fixes result in RLTweaker being updated to Ice & Fire 2.0 (of the Ice & Fire: RLCraft Edition fork), which already includes the majority of the fixes for Ice & Fire contained in RLTweaker. As a result, I have removed these associated fixes. Furthermore, since Ice & Fire 2.0 includes changes to the stoned mob API, we have also updated the stoned mob checks to utilize this updated API.

Meldexun commented:
I'm pretty sure the stoned kobold fix is unnecessary because I have fixed this issue in LycanitesMobs but I haven't checked. If that is the case, then the IceAndFireHandler class and IceAndFireReflector class have to be removed as well.

Also, all IceAndFire-related config options should be removed. This probably only includes some fields of the PatchConfig class and the IceAndFireConfig class (and wherever it is used).

kotlin-programmer commented:
> I'm pretty sure the stoned kobold fix is unnecessary because I have fixed this issue in LycanitesMobs but I haven't checked. If that is the case, then the IceAndFireHandler class and IceAndFireReflector class have to be removed as well.
> 
> Also, all IceAndFire-related config options have to be removed. This probably only includes some fields of the PatchConfig class and the ConfigIceAndFire class (and wherever it is used).

I have verified that this patch was incorporated into the Ice & Fire fork, and therefore is unnecessary here. As a result, I've removed all of the Ice & Fire handling / tweaks. Well, withstanding the generic tweaks the additionally apply to Ice & Fire mobs, such as the entity collision and AI tweaks.

Meldexun commented:
I assume `EntitySeaSerpentBubbles` doesn't need the less collisions patch of RLTweaker, right?

The `ConfigIceAndFire` and `IceAndFireReflect` classes are not deleted yet.
